File: sortinplace/QuickSort.java
package sortinplace;
import java.util.concurrent.RecursiveAction;

abstract class QuickSort extends RecursiveAction implements ISwapper {
    protected final int low,high;
    protected final int[] array;

    public QuickSort(int[] array,int low,int high){
        this.array = array;
        this.low = low;
        this.high = high;
    }
    protected abstract int partition();
    protected abstract void sort();
    @Override
    protected void compute(){
        this.sort();
    }
}

File: src/algorithms/sort/quick/QuickSort.php
<?php
declare(strict_types=1);
namespace ZimCodes\Self_Learning\Algorithms\Sort\Quick;
use ZimCodes\Self_Learning\Algorithms\Sort\SortIn;
use ZimCodes\Self_Learning\Algorithms\SwapperTrait;

abstract class QuickSort extends SortIn{
    use SwapperTrait;
    abstract protected function sorting(array &$array,int $low=0,int $high=0): void;
    abstract protected function partition(array &$array,int $low,int $high):int;
}